### DISCUSSION TITLE
REST API allows us to set deployment environment "selected branches", but not apply deployment rules

### DISCUSSION THREAD

**Discussion by krisrecur** [community/community#14190] [Actions]
REST API allows us to set environment "selected branches", but not apply the all important deployment rules

<img width="1596" alt="Deployment branches" src="https://user-images.githubusercontent.com/101663319/161436354-e3f33578-c6c2-4e51-a884-848b0e5166c8.png">

API docs are not clear

https://docs.github.com/en/enterprise-cloud@latest/rest/reference/deployments#environments

After creating a deployment branch rule via the web UI, I see the following in the JSON output of the environment view operation

```
  "protection_rules": [
    {
      "id": 99999,
      "node_id": "XXXXXXXXXXXXXX",
      "type": "branch_policy"
    }
  ],
```

**Comment by ethomson** [comment]
Thanks for the feedback @krisrecur - I'll put this on the backlog for future improvements.

**Comment by matsest** [comment]
Any news on triage of this in the backlog @ethomson? Or if you can mention someone else that's involved in this backlog.

Btw, should this be labeled with API?

**Comment by StephenWithPH** [comment]
From https://docs.github.com/en/rest/deployments/environments#create-or-update-an-environment:

>Note: Although you can use this operation to specify that only branches that match specified name patterns can deploy to this environment, you must use the UI to set the name patterns.

I agree that it would be useful for the API to be extended to handle programmatically setting the branches.

This has come up here: https://github.com/integrations/terraform-provider-github/issues/922#issuecomment-1011908557

**Comment by matsest-klp** [comment]
Is this now supported? :) - see https://docs.github.com/en/rest/deployments/branch-policies#create-a-deployment-branch-policy
